File: allLongestStrings.js
function allLongestStrings(inputArray) {
    var output = [];
    var longest = 0;
    // find longest length
    for (var i = 0; i < inputArray.length; i++) {
        if (longest < inputArray[i].length) longest = inputArray[i].length;
    }
    for (var i = 0; i < inputArray.length; i++) {
        if (inputArray[i].length === longest) {
            output.push(inputArray[i]);
        }
    }
    return output;
}

File: 1_intro/3_smooth_sailing/9_allLongestStrings/allLongestStrings.lua
function allLongestStrings(inputArray)
  local countedArray = {}
  local longest = 1
  for i = 1, table.getn(inputArray) do
    local key = string.len(inputArray[i])
    if key >= longest then
      longest = key
      if countedArray[key] == nil then
        countedArray[key] = { inputArray[i] }
      else
        table.insert(countedArray[key], inputArray[i])
      end
    end
  end
  return countedArray[longest]
end